mp/src/game/client/momentum/ui/HUD/hud_mapfinished.cpp

```diff
@@ -245,6 +259,12 @@ bool CHudMapFinishedDialog::ClosePanel()
     if (!m_bCanClose && !m_bIsGhost)
         return false;
 
+    // If player gets a new pb, update player best
+    if (m_flOverallDiff < 0.00f)
```

Comment:
`m_flOverallDiff` and `m_flPbOverallTime` are not initialized in the constructor, meaning that this check may fail. It may be good to initialize both of them to `0.0f`.